conda_libmamba_solver/shards.py

```diff
@@ -571,6 +597,7 @@ def fetch_shards_index(
             # load from cache without network request
             shards_data = repo_cache.cache_path_shards.read_bytes()
 
+        # If we don't have shards_data yet, try fetching (repodata_shards handles offline mode)
```

Comment:
repodata_shards() could be refactored, at the moment there isn't an elegant way for it to skip to returning the cached shard index. I think it should check cache_state.has_format("shards"), if that's False fall back to the non-shards offline support, if that's True load the cache without doing networking or checking for expiry.

https://github.com/conda/conda/pull/15471 begins trying to improve the recent binary cache API invented for sharded repodata so that there is a method call and not reading the info.json directly.